localization/ekf_localizer/include/ekf_localizer/ekf_localizer.hpp

```diff
@@ -138,16 +153,18 @@ class EKFLocalizer : public rclcpp::Node
   };
 
   /* for model prediction */
-  geometry_msgs::msg::TwistStamped::SharedPtr
-    current_twist_ptr_;                                          //!< @brief current measured twist
-  geometry_msgs::msg::PoseStamped::SharedPtr current_pose_ptr_;  //!< @brief current measured pose
-  geometry_msgs::msg::PoseStamped current_ekf_pose_;             //!< @brief current estimated pose
+  std::queue<TwistInfo> current_twist_info_queue_;    //!< @brief current measured pose
+  std::queue<PoseInfo> current_pose_info_queue_;      //!< @brief current measured pose
+  geometry_msgs::msg::PoseStamped current_ekf_pose_;  //!< @brief current estimated pose
   geometry_msgs::msg::PoseStamped
     current_ekf_pose_no_yawbias_;  //!< @brief current estimated pose w/o yaw bias
   geometry_msgs::msg::TwistStamped current_ekf_twist_;  //!< @brief current estimated twist
   std::array<double, 36ul> current_pose_covariance_;
   std::array<double, 36ul> current_twist_covariance_;
 
+  int pose_smoothing_steps_;
+  int twist_smoothing_steps_;
```

Comment:
Ideally, these values should be dynamic for the input covariance. Large smoothing_steps for small covariance.
We can leave this feature for future work.

But wouldn't it be better to have these parameters in Pose/TwistInfo for future expansion?

Reply:
I agree that, ideally speaking, these values should be dynamic. I added these parameters in PoseInfo & TwistInfo.
https://github.com/autowarefoundation/autoware.universe/pull/1027/commits/6ef59776db89f9220e4d9d1919d3510c87c6ae46